.circleci/config.yml

```diff
@@ -1,17 +1,141 @@
 version: 2.1
 
-step-library:
-  - &restore-cache
-      restore_cache:
-        keys:
-          - nav-cache-v{{ .Environment.CARTHAGE_CACHE_VERSION }}-{{ .Environment.CIRCLE_JOB }}-{{ checksum "Cartfile.resolved" }}
+commands:
+  restore-carthage-cache:
+    parameters:
+      xcode:
+        type: string
+    steps:
+      - restore_cache:
+          key: carthage-cache-v1-{{ checksum "Cartfile.resolved" }}-<< parameters.xcode >>
 
-  - &save-cache
-      save_cache:
-        key: nav-cache-v{{ .Environment.CARTHAGE_CACHE_VERSION }}-{{ .Environment.CIRCLE_JOB }}-{{ checksum "Cartfile.resolved" }}
-        paths:
-          - Carthage
+  save-carthage-cache:
+    parameters:
+      xcode:
+        type: string
+    steps:
+      - save_cache:
+          key: carthage-cache-v1-{{ checksum "Cartfile.resolved" }}-<< parameters.xcode >>
+          paths:
+            - Carthage
+
+  save-api-diff-cache:
+    parameters:
+      key:
+        type: string    
+      is_template:
+        type: boolean
+        default: false
+    steps:
+      - when:
+          condition: << parameters.is_template >>
+          steps: 
+            - save_cache:
+                key: nav-sdk-diff-cache-v5-{{ << parameters.key >> }}
+                paths:
+                  - api_logs
+      - unless:
+          condition: << parameters.is_template >>
+          steps:
+            - save_cache:
+                key: nav-sdk-diff-cache-v5-<< parameters.key >>
+                paths:
+                  - api_logs
+
+  restore-api-diff-cache:
+    parameters:
+      key:
+        type: string
+      is_template:
+        type: boolean
+        default: false
+    steps:
+      - when:
+          condition: << parameters.is_template >>
+          steps: 
+            - restore_cache:
+                key: nav-sdk-diff-cache-v5-{{ << parameters.key >> }}
+      - unless:
+          condition: << parameters.is_template >>
+          steps:
+            - restore_cache:
+                key: nav-sdk-diff-cache-v5-<< parameters.key >>
 
+  build-api-diff-cli-tool:
+    description: "Builds CLI tool which produce diff reports"
```

Comment:
Can we distribute `swift-api-diff` as a prebuild CLI so we don't have to build it each time?

Reply:
This will also allow to remove the steps that saves and restores the binary

Reply:
- Can be a subsequent improvement. Just one note: it is not built each time, it is built only once per released version.

Reply:
Feel free to post the suggestion in https://github.com/mapbox/swift-api-diff repo for future tracking or anywhere where it is more appropriate.